.github/workflows/manual-deployment.yml

```diff
@@ -0,0 +1,51 @@
+name: Manual deployments for all envs
+
+on:
+  workflow_dispatch:
+    inputs:
+      spaName:
+        description: Please enter the SPA name to deploy (eg. component-catalog).
+        default: 'home'
+        required: true
+        type: string
+      env:
+        description: Select then env you want to deploy your SPA.
+        options:
+          - dev
+          - qa
+          - stage
+          - prod
+        default: dev
+        required: true
+        type: choice
+
+env:
+  NODE_VERSION: 16
+  SPA_NAME: "${{ inputs.spaName }}-spa"
+  SPASHIP_DEPLOY_URL: "${{ secrets.SPASHIP_DEPLOY_URL }}${{ inputs.env }}"
+
+jobs:
+  build:
+   runs-on: self-hosted
```

Comment:
Can you split this job into 2, in order to optimize it.
Since we don't need the self-hosted runner for building the app. Just for deploying.
So run the build on github shared runner, and the deploy on self-hosted.

Reply:
@deshmukhmayur the SPA needs to be built on the self hosted container as it needs the env files which are also internal

Reply:
Oh right. Nevermind that then.